frontend/src/lib/utils/onKeyboardFocusOut.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * An action that adds an event handler that fires when a keyboard user's
+ * focus moves out from the element or any of its descendants.
+ *
+ * Due to different browser behaviours, we have to do the detection in a
+ * bit of a contrived manner:
+ * - When the users moves the focus within the focusable elements of the
+ *   container, the following events are fired: `keydown` (Tab),
+ *   `focusout`, `focusin` and `keyup` (Tab).
+ * - If the focus moves out of the container, only the first two are
+ *   fired.
+ * - We listen for the `keydown` (Tab) event and then set a timeout to
+ *   check whether we have received focus, and if not, we assume the focus
+ *   has left the container. We listen to `focusout` and `focusin` to
+ *   track whether we have focus or not. (We cannot use
+ *   `document.activeElement` because it's not reliable.)
+ *
+ * @param node The target element
+ * @param callback The event handler to call
+ *
+ * @example `<Navigation on:keyboardFocusOut={closeDrawer} />`
+ */
+export function onKeyboardFocusOut(node: HTMLElement, callback: () => void) {
```

Comment:
This looks a bit hacky. We can revisit this in the future and find other alternatives.

Reply:
I agree, it's quite contrived. I spent quite some time trying come up with a nicer solution, but this was the only working one I could find. One tricky issue is that when tabbing through the navigation menu, keyboard focus will move out of the page after you reach the last nav item. Thus, we won't have any focused element on the page anymore…